lib/ret/asset.ex

```diff
@@ -33,6 +40,13 @@ defmodule Ret.Asset do
     Repo.transaction(multi)
   end
 
+  def asset_by_sid_for_account(account, asset_sid) do
+    from(a in Asset,
+      where: a.asset_sid == ^asset_sid and a.account_id == ^account.account_id,
+      preload: [:account, :asset_owned_file, :thumbnail_owned_file])
```

Comment:
should check `state` is active I think

Reply:
On the owned files? Asset doesn't have an active field, I thought we had settled on actually deleting assets/project/project_assets instead of using an active field?

Reply:
ah if we left that column out that is fine. in general i've been adding the state column pre-emptively in the data model (but don't really care that much if we do it later.) my past experience is that doing it up-front can be valuable so you don't miss places later if/when you need to add a soft delete or other filtering, but not a big deal to leave it alone for now.